### ISSUE TITLE
Feature Logging

### ISSUE THREAD

**Issue by nathanjent** [rsanchez-wsu/fa15-ceg3120#8]
We need a common interface to log debug and error messages. These would need to be written to the database since I believe the server GUI team wants to display log messages in a table.
I suggest using something like [log4j](http://logging.apache.org/log4j/2.x/).

**Comment by rsanchez-wsu** [comment]
@nathanjent I think that [slf4j](http://www.slf4j.org) is a better choice.  It uses the façade pattern and allows you to have any of a number of different logging frameworks perform the actual logging.  We can use slf4j as the interface and then lgo4j as the actual implementation.  Feel free to go ahead and add them to the Ivy dependencies for the project.  @Shukaro can help with that if needed.

**Comment by nathanjent** [comment]
I started implementing the logging with slf4j and log4j as the framework. I added the dependencies to ivy.xml. I also added a target to common.xml to generate the log4j.xml in each module. However, when I tried to run the ant compile target for the whole project it only created the log4j.xml in the contractor_client directory and didn't iterate to the others. Is the build working for you, @Shukaro.
